lib/trino-filesystem/src/main/java/io/trino/filesystem/cache/CacheInputFile.java

```diff
@@ -64,7 +67,14 @@ public TrinoInputStream newStream()
     public long length()
             throws IOException
     {
-        return delegate.length();
+        if (length.isEmpty()) {
+            Optional<String> key = keyProvider.getCacheKey(delegate);
```

Comment:
maybe:
```
length = OptionalLong.of(keyProvider.getCacheKey(delegate)
   .map(key -> cache.cacheLength(delegate, key))
   .orElse(delegate.length()))
```

Reply:
Putting `cacheLength` in lambda requires try/catch block around it, so it ends up becoming more verbose than now

Reply:
cacheLength throws checked exception